### DISCUSSION TITLE
atomic chezmoi re-add?

### DISCUSSION THREAD

**Discussion by danielo515** [twpayne/chezmoi#2576] [Q&A] [answered]
Hello. I wanted a flow to re-add a single file that I edited outside of chezmoi (I have some good reasons to do it).
I saw #1598, there it is mentioned that `chezmoi add` will not overwrite the changes of a file, and that I have to do it with `chezmoi add -f`.
Is that the official way to do it? Will it be possible to make chezmoi re-add accept a file path so it is more specific about  your intentions? Just like git added `switch` to change between branches even though you can do it with checkout just fine.

**Comment by halostatue** [comment, marked as answer]
I tend to use `chezmoi merge <FILE>` or `chezmoi merge-all` for when I edit files outside of chezmoi.

**Reply by danielo515** [reply]
I want to do it in an automated fashion, do those files require interaction?
In my case, this files are only supposed to be edited outside of chezmoi folder, so it should be safe to add them, but is good to know about merge

**Reply by twpayne** [reply]
Yes, for an unconditional re-add of a known file, use `chezmoi add --force $FILENAME`.

`chezmoi re-add` is more the for case where you're not sure what files you've changed locally.

**Reply by danielo515** [reply]
Ok, thanks for confirmation. Maybe a little annotation can be added to the `re-add` docs? Like, for readding individual files, please use `chezmoi add --force $FILENAME`

**Reply by twpayne** [reply]
Done in 8a9399b48a8b45afe66a76f9c45f4d7a1ba99fe4.
